packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -118,6 +121,17 @@ function BlockSwitcherDropdownMenuContents( {
 			</p>
 		);
 	}
```

Comment:
I'm wondering if we could get `isUsingBindings` from this `useSelect`. Something similar to what is being done [here](https://github.com/WordPress/gutenberg/pull/62317/files#diff-9ac2a4466e7c0a85aaff0b40c5d7bc64c894132528ca3c879db4bebb438617dcR99-R100).

If we do this, I believe we could avoid passing `isUsingBindings` to `BlockSwitcher` and `BlockSwitcherDropdownMenuContents` components.

Additionally, I think we wouldn't need `isUsingBindings` in the block toolbar anymore ([here](https://github.com/WordPress/gutenberg/pull/62317/files#diff-9ac2a4466e7c0a85aaff0b40c5d7bc64c894132528ca3c879db4bebb438617dcR179-R183)) and we could just have something like `hasPatternOverrides`, as it is now specific to pattern overrides, right?

I am not 100% sure if it makes sense, but maybe it is worth taking a look.

Reply:
I believe so, yes! However I don't think it makes much difference. Currently we still need `isUsingBindings` in block toolbar to apply the `is-connected` class name. We can somehow move the class name to block switcher, but in terms of performance I don't think it matters that much. Do you think it's worth exploring?

Reply:
I wasn't thinking about performance but about readability, but I don't have a strong opinion, so whatever you decide it's okay for me 🙂

Reply:
I think it reads okay for me, but I'm obviously biased 😅. Happy to do some further improvements if we find something confusing later! 🙇